### ISSUE TITLE
Polling should be case-insensitive

### ISSUE THREAD

**Issue by tombuildsstuff** [Azure/go-autorest#195]
:wave:

We're provisioning a Search Service through the Azure SDK for Go - and have noticed since ~mid July that the deployment times have been increasing.

Upon further investigation - it appears that AutoRest doesn't handle the `provisioningState` field being lower-case - as is the case in the Search Service API:

After sending this request:
```
{
	"location": "westeurope",
	"tags": {
		"database": "test",
		"environment": "staging"
	},
	"properties": {},
	"sku": {
		"name": "standard"
	}
}
```

and polling:

```
{
	"id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/tharvey-dev2/providers/Microsoft.Search/searchServices/tharveysearch",
	"name": "tharveysearch",
	"type": "Microsoft.Search/searchServices",
	"location": "West Europe",
	"tags": {
		"database": "test",
		"environment": "staging"
	},
	"properties": {
		"replicaCount": 1,
		"partitionCount": 1,
		"status": "provisioning",
		"statusDetails": "",
		"provisioningState": "provisioning",
		"hostingMode": "Default"
	},
	"sku": {
		"name": "standard"
	}
}
```

which, eventually returns the following:

```
{
	"id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/tharvey-dev2/providers/Microsoft.Search/searchServices/tharveysearch",
	"name": "tharveysearch",
	"type": "Microsoft.Search/searchServices",
	"location": "West Europe",
	"tags": {
		"database": "test",
		"environment": "staging"
	},
	"properties": {
		"replicaCount": 1,
		"partitionCount": 1,
		"status": "running",
		"statusDetails": "",
		"provisioningState": "succeeded",
		"hostingMode": "Default"
	},
	"sku": {
		"name": "standard"
	}
}
```

However the SDK continues polling indefinitely (see below, the first `successful` response is highlighted):

<img width="314" alt="screen shot 2017-11-13 at 15 44 51" src="https://user-images.githubusercontent.com/666005/32734412-dbe54aae-c889-11e7-9411-191354bbbc75.png">

Upon updating [this line](https://github.com/Azure/go-autorest/blob/master/autorest/azure/async.go#L39) to be lower-case and rebuilding - the SDK appears to work as expected.

As such - would it be possible to update AutoRest to allow this field to be case insensitive?

On a related note - it appears the Portal is shying users away from creating new Search Services - is it Deprecated / EOL'd? Screenshot from the Portal - which appears to be applicable in every region:

<img width="287" alt="screen shot 2017-11-13 at 15 13 02" src="https://user-images.githubusercontent.com/666005/32734447-f546d4e0-c889-11e7-85ad-7a4750a1095b.png">

Thanks!

**Comment by jhendrixMSFT** [comment]
This is fixed in v9.4.1.  @joshgav will follow up on your questions regarding search services.

**Comment by tombuildsstuff** [comment]
Awesome thanks @jhendrixMSFT :)
